src/main.cpp

```diff
@@ -875,6 +875,16 @@ int main(int argc, char* argv[])
     return 0;
   }
 
+  if (args.build_mode == BuildMode::AHEAD_OF_TIME) {
+    auto num_cpus = bpftrace.get_num_possible_cpus();
+    if (num_cpus > 1024) {
+      LOG(WARNING) << "Detected " << num_cpus
+                   << " cpus. For ahead-of-time compilation there is a max of "
+                      "1024 cpus so there may be incorrect data for 'count' "
+                      "and 'sum' aggregations.";
+    }
+  }
```

Comment:
I think this warning is only relevant if the _runtime_ host has >1024 CPUs. So if build host had >1024 CPUs but runtime had <1024, it'd still be ok. So I think this should go in (or near) `aot_main.cpp`. 

Having this hard coded message is probably OK if you plan on removing it soonish. Was thinking if it makes sense to analyze RequiredResources in aot.cpp::load_required_resources. It'd be more accurate -- we could check if count() or sum() is even being used. IMO it'd be a little cleaner to have it there. Up to you.

Reply:
> So I think this should go in (or near) aot_main.cpp.

Ah right. Good catch. I'm still getting familiar with AOT :) 

> we could check if count() or sum() is even being used. IMO it'd be a little cleaner to have it there. Up to you.

I plan on not having this hack for too long so the less additional logic the better - besides are we really worried about the case where someone is running this on a machine with over 1024 cores?

Reply:
> I plan on not having this hack for too long so the less additional logic the better - besides are we really worried about the case where someone is running this on a machine with over 1024 cores?

I think it's reasonable to be worried about this case. Looks like we are getting users working on supercomputers now